packages/discord.js/src/structures/ModalSubmitFieldsResolver.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const { ComponentType } = require('discord-api-types/v9');
+const { TypeError } = require('../errors');
+
+class ModalSubmitFieldsResolver {
+  constructor(components) {
+    /**
+     * The components within the modal
+     * @type {Array<ActionRow<ModalFieldData>>} The components in the modal
+     */
+    this.components = components;
+  }
+
+  /**
+   * The extracted fields from the modal
+   * @type {ModalFieldData[]} The fields in the modal
+   */
+  get fields() {
```

Comment:
This is just me throwing an idea, but couldn't we just convert fields to a static property of the class (consistency with CommandOptions one)